packages/example-log-extension/src/providers/log-action.provider.ts

```diff
@@ -8,19 +8,28 @@ import {CoreBindings} from '@loopback/core';
 import {OperationArgs, ParsedRequest} from '@loopback/rest';
 import {getLogMetadata} from '../decorators/log.decorator';
 import {EXAMPLE_LOG_BINDINGS, LOG_LEVEL} from '../keys';
-import {LogFn, TimerFn, HighResTime, LevelMetadata} from '../types';
+import {
+  LogFn,
+  TimerFn,
+  HighResTime,
+  LevelMetadata,
+  LogWriterFn,
+} from '../types';
 import chalk from 'chalk';
-import * as debugModule from 'debug';
-const debug = debugModule('loopback:example:extension:log');
 
 export class LogActionProvider implements Provider<LogFn> {
+  // LogWriteFn is an optional dependency and it falls back to `logToConsole`
+  @inject(EXAMPLE_LOG_BINDINGS.LOGGER, {optional: true})
+  private logWriter: LogWriterFn = logToConsole;
+
+  @inject(EXAMPLE_LOG_BINDINGS.APP_LOG_LEVEL, {optional: true})
+  private logLevel: number = LOG_LEVEL.WARN;
```

Comment:
Both `logWriter` and `logLevel` must be public, so that consumers of this class can customize them explicitly (without depending on Dependency Injection).